StripeUICore/StripeUICore/Source/Elements/TextField/TextFieldElement.swift

```diff
@@ -176,4 +176,12 @@ extension TextFieldElement: TextFieldViewDelegate {
         isEditing = view.isEditing
         delegate?.continueToNextField(element: self)
     }
+
+    func traitCollectionDidChangeForTextField(
+        view: TextFieldView,
+        traitCollection: UITraitCollection
+    ) {
+        view.updateUI(with: viewModel)
+        delegate?.didUpdate(element: self)
+    }
```

Comment:
This is the best solution I came up with, it is the same pattern as when the text changes.

Reply:
I was thinking about overriding this in the accessory view itself - I guess in the CVC case, it'd be a custom UIView/UIImageView subclass that knows its light and darkmode images and overrides `traitCollectionDidChange` to update its image accordingly.

Reply:
There are some issues with that approach:
- The view's brightness depends on the theme's brightness, not the OS's brightness, so just updating the accessory view based on the new system brightness won't work, we need an updated theme. 
- The accessory view doesn't have access to the theme directly, because the theme is owned by the view model, as is the accessory view, hence why we need to regenerate the viewModel and update the view with it.

One thing I did notice while looking at this is that we shouldn't just be looking at the theme's overall brightness, we should be looking at the field's brightness, like this case for example:

<img width="413" alt="Screen Shot 2022-11-16 at 1 34 21 PM" src="https://user-images.githubusercontent.com/114543737/202299941-3ce28668-e697-4b54-9bc4-be4c4c890c01.png">

The theme is bright, but the field is dark, we could probably use the same brightness as the text for the icons.